src/frontend/src/components/common/RichText/index.tsx

```diff
@@ -0,0 +1,249 @@
+/******************************************************************************  
+ *   MODULE: RichText -> Rich Text Editor
+ *   AUTHOR: BC GOV
+ *   DATE:   20th November 2022   
+ *   LAST UPDATED:
+ * 
+ *   DESCRIPTION: This module enable a rich text feature for forms. It can
+ *                add Bold, underline, italics, sub-script, superscript, 
+ *                ordered list, unordered list and clear formatting to default. 
+ * 
+ *   HOW TO USE THIS MODULE: 
+ *   Input - This module accepts a few data elements from the parent component
+ *           - [name] - For each instance of this module a unique 
+ *                      name must be assigned
+ *           - [value] - This allows two ways databinding. 
+ *                 Use Case 1: If the form needs to be populated with 
+ *                       exiting text from a database for example 
+ *                       during initialization.
+ *                 Use Case 2: If the description is changed in another component
+ *                       the changes are tracked and will populate the richText
+ *                       with this value. This usecase is rare.
+ *                               
+ *   Output[HTML] - This module will ouput the content of the textarea in 
+ *                  HTML format to the parent using the onChange parent function 
+ *           - [onChange] - For every keyup event the onChange function is called
+ *        
+ *   HOW TO TEST THIS MODULE: 
+ *   This module is tightly coupled with the DPIA project and its parent 
+ *   components                       
+ *   !TODO - Write test cases to validate input and output
+ ******************************************************************************
+ */
+
+import React from 'react';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { solid, regular, brands, icon } from '@fortawesome/fontawesome-svg-core/import.macro'
+import {
+    faBold, faItalic, faUnderline, faSubscript,
+    faSuperscript, faList, faListOl, faParagraph
+} from '@fortawesome/free-solid-svg-icons';
+
+class RichText extends React.Component {

```

Comment:
Why are we using a class based component here? The rest of the frontend in function based.

Reply:
i strong suggest we use func based component

Reply:
most of the documentation is class based. Just followed the document. In future interactions i can update it.